### DISCUSSION TITLE
cause of non-convergence in SCPH calculation.

### DISCUSSION THREAD

**Discussion by Gi-HoonKang** [ttadano/alamode#90] [Q&A]
Hello. I'm sorry to bother you, but I have a question about the cause of non-convergence in SCPH calculation.
I obtained non-convergence result from my SCPH calculation. The input was as below.

&general
 PREFIX = WH-sc4-V1H2-meta_scph
 MODE = SCPH
 NKD = 2; KD = W H
 FCSXML = WH-sc4-V1H2-meta-random.xml
 TMIN =0; TMAX = 2000; DT = 100
/

&scph
 SELF_OFFDIAG = 0
 MAXITER = 500
 MIXALPHA = 0.2
 KMESH_INTERPOLATE = 1 1 1  #same with the supercell size
 KMESH_SCPH = 1 1 1  # should be multiple of KMESH_INTERPOLATE
/

&cell
  1.8897259886 # factor in Bohr unit
  1.2686920000000001e+01 0.0 0.0 # a1
  0.0 1.2686920000000001e+01 0.0 # a2
  0.0 0.0 1.2686920000000001e+01 # a3
/

&kpoint
 2
 1 1 1
/

I already knew that decreasing "MIXALPHA" or "DT" can improve the convergence from ALAMODE seminar. However, I am also curious about the other reasons. Is there any possibilities that inaccurate force constants or low training dataset (related to compressive sensing method) cause the non-convergence? 
I attached the output of SCPH calculation just in case. I always appreciate your help.

[scph.log](https://github.com/ttadano/alamode/files/9503084/scph.log)

**Comment by ttadano** [comment]
The bad convergence may result from the inaccuracy of fourth-order IFCs. I would suggest increasing the training data for compressive sensing and checking if the convergence rate improves.

From your scph output, the calculation converged at 0 K and 200 K, but not at other temperatures. If you set `LOWER_TEMP = 0` in the &scph field, the convergence can improve hopefully.

**Comment by Lamed-git** [comment]
I recently met a problem with non-convergence in SCPH calculations too. I double-checked the input file and found that I had written the wrong cell parameters when generating the force constants with alm program, so I hope you can check the correctness of the force constant file.
